snapcraft/internal/lxd.py

```diff
@@ -141,7 +140,7 @@ def execute(self, step='snap', args=None):
             command = ['snapcraft', step]
             if step == 'snap':
                 command += ['--output', self._snap_output]
-            if self._host_arch != self._project_options.deb_arch:
+            if self._project_options.target_arch_specified:
```

Comment:
Looking at this further, why aren't we just passing the cli args used to actually call snapcraft on to the container? Why try to reproduce them by looking back at the options, which essentially represent the already-parsed arguments?